plugin/core/sessions.py

```diff
@@ -979,6 +974,18 @@ def should_notify_did_save(self) -> Tuple[bool, bool]:
     def should_notify_did_close(self) -> bool:
         return self.capabilities.should_notify_did_close()
 
+    # --- FileWatcherProtocol ------------------------------------------------------------------------------------------
+
+    def on_file_event_async(self, events: List[FileWatcherEvent]) -> None:
+        changes = []  # type: List[FileEvent]
+        for event in events:
+            kind, filepath = event
+            changes.append({
+                'uri': filename_to_uri(filepath),
+                'type': file_watcher_kind_to_lsp_file_change_type(kind),
+            })
+        self.send_notification(Notification.didChangeWatchedFiles({'changes': changes}))
```

Comment:
If we have a file watcher installed,
then the LSP client will say that it support dynamicRegistration.
```   
if get_file_watcher_implementation():
    text_document_capabilities["didChangeWatchedFiles"] = {"dynamicRegistration": True}
```

> It is recommended that servers register for these file system events using the registration mechanism. _In former implementations clients pushed file events without the server actively asking for it._

Will the LSP client send `didChangeWatchedFiles` notifications to severs who are not interested in these type of notifications? 

My answer to that would be only if the server dynamically registered to that capability. If the server didn't registered than I would guess that dynamic file watchers would [not get instancated](https://github.com/sublimelsp/LSP/pull/1764/files#diff-8fe432afb96c719bc734bcb2a1045d852979d09ea395956f1b979861432c2b05R1293-R1294).
If file watchers are not instantiated than `on_file_event_async` would never be called, and the server will never be notified with the didChangeWatchedFiles notification.

... Sorry for these kind of questions, I am just trying to grasp this part of the spec :)

Reply:
It's like you said.

For servers that don't dynamically register the watcher but want to be notified about some file changes (like vetur or eslint), we need to define the configuration in their ClientConfig.